awx/ui_next/src/screens/Project/ProjectList/ProjectList.test.jsx

```diff
@@ -0,0 +1,222 @@
+import React from 'react';
+import { ProjectsAPI } from '@api';
+import { mountWithContexts, waitForElement } from '@testUtils/enzymeHelpers';
+
+import ProjectsList, { _ProjectsList } from './ProjectList';
+
+jest.mock('@api');
+
+const mockProjects = [
+  {
+    id: 1,
+    name: 'Project 1',
+    url: '/api/v2/projects/1',
+    type: 'project',
+    scm_type: 'git',
+    summary_fields: {
+      last_job: {
+        id: 9000,
+        status: 'successful',
+      },
+      user_capabilities: {
+        delete: true,
+        update: true,
+      },
+    },
+  },
+  {
+    id: 2,
+    name: 'Project 2',
+    url: '/api/v2/projects/2',
+    type: 'project',
+    scm_type: 'svn',
+    summary_fields: {
+      last_job: {
+        id: 9002,
+        status: 'successful',
+      },
+      user_capabilities: {
+        delete: true,
+        update: true,
+      },
+    },
+  },
+  {
+    id: 3,
+    name: 'Project 3',
+    url: '/api/v2/projects/3',
+    type: 'project',
+    scm_type: 'insights',
+    summary_fields: {
+      last_job: {
+        id: 9003,
+        status: 'successful',
+      },
+      user_capabilities: {
+        delete: false,
+        update: false,
+      },
+    },
+  },
+];
+
+describe('<ProjectsList />', () => {
+  beforeEach(() => {
+    ProjectsAPI.read.mockResolvedValue({
+      data: {
+        count: mockProjects.length,
+        results: mockProjects,
+      },
+    });
+
+    ProjectsAPI.readOptions.mockResolvedValue({
+      data: {
+        actions: [],
+      },
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('initially renders successfully', () => {
+    mountWithContexts(
+      <ProjectsList
+        match={{ path: '/projects', url: '/projects' }}
+        location={{ search: '', pathname: '/projects' }}
+      />
+    );
+  });
+
+  test('Projects are retrieved from the api and the components finishes loading', async done => {
```

Comment:
:+1: This is important to test - in a lot of cases these list components can _technically_ render but would actually fail to initialize since they haven't finished all of their async things.